samples/schema-registry-integration/README.adoc

```diff
@@ -0,0 +1,133 @@
+== Spring Cloud Stream and Schema Evolution in Action with Kafka Binder.
+
+These are a set of Spring Boot applications to demonstrate Schema Evolution using Spring Cloud Stream with Kafka binder.
+Producer V1 (`producer1`), Producer V2 (`producer2`), and Consumer (`consumer`) are included in this project.
+
+=== Requirement
+As a developer, I'd like to design my consumer to be resilient to differing payload schemas.
+
+=== Assumptions
+For this demonstration, we will simply assume there are two producers producing events with different payload schemas.
+A consumer that consumes both the payload versions will be designed to adapt to evolving schemas.
+
+These applications integrate with Spring Cloud Stream Scheam Registry.
+Both producers and consumers interact with this Schema Registry to register and evolve the schema.
+
+[[build-apps]]
+=== Building the applications
+To build the applications simply execute the following command from the `schema-registry-integration` directory:
+[source,bash]
+----
+./mvnw clean install
+----
+NOTE: The apps can be built and run from w/in an IDE (such as IntelliJ) but you will need to invoke the Maven `package` goal and then `refresh` the project as the Avro Maven plugin needs to execute so that it generates the required model classes - otherwise you will see compile failures around missing `Sensor` class.
+
+[[run-apps]]
+=== Running the applications
+
+==== Pre-requisites
+****
+* The components have all been built by following the <<build-apps>> steps.
+* Apache Kafka broker available at `localhost:9092`
+
+TIP: The included link:../../../tools/kafka/docker-compose/README.adoc#_all_the_things[Kafka tools] can be used to easily start a broker locally on the required coordinates
+****
+
+If you want to use schema registry with Postgres or MYSQL, either start them locally or use the provided docker-compose scripts above.
+
+==== Steps
+Make sure the above pre-requisites are satisfied and that you are in the `schema-registry-integration` directory and follow the steps below.
+
+- Start the Schema Registry server (Change these commands accordingly if you are not on a Unix like platform)
+[source,bash]
+----
+java -jar <ROOT-OF-SPRING-CLOUD-STREAM-CHECKOUT>/schema-registry/spring-cloud-stream-schema-registry-server-<VERSION>.jar
+----
+
+This starts the schema registry with a local H2 database.
+
+To use Postgres database instead of H2, set the following property :
```

Comment:
I would reword slightly to say 
```
To use Postgres instead of H2, additional properties must be specified when starting the server:
```

Same for MySQL instructions below.